h5py/_hl/vds.py

```diff
@@ -123,7 +123,7 @@ def __init__(self, shape, dtype=None, maxshape=None):
         self.sources = []
 
     def __setitem__(self, key, source):
-        sel = SimpleSelection(self.shape)[key]
+        sel = select(self.shape, key, source.sel.id)
```

Comment:
I'm pretty sure passing `source.sel.id` here is wrong. It's only used if we're selecting with a `RegionReference` object, which I haven't used. But I think setting it to `None`, as you have for the source selection, is probably appropriate. That will just mean that we can't use a `RegionReference` for now.